test/helpers/user.ts

```diff
@@ -1,78 +1,85 @@
 import { User } from "@services/mongoose/schema"
 import { toObjectId } from "@services/mongoose/utils"
 
 const users = UsersRepository()
 
-export const getPhoneByTestUserIndex = (index: number) => {
-  const entry = yamlConfig.test_accounts[index]
+const getPhoneByTestUserIndex = (ref: string) => {
+  const entry = yamlConfig.test_accounts.find((item) => item.ref === ref)
   const phone = entry.phone as PhoneNumber
   return phone
 }
 
-export const getUserByTestUserIndex = async (index: number) => {
-  const phone = getPhoneByTestUserIndex(index)
+const getUserByTestUserRef = async (ref: string) => {
+  const phone = getPhoneByTestUserIndex(ref)
   const user = await UsersRepository().findByPhone(phone)
   if (user instanceof Error) throw user
   return user
 }
 
-export const getAccountByTestUserIndex = async (index: number) => {
-  const user = await getUserByTestUserIndex(index)
+export const getAccountByTestUserRef = async (ref: string) => {
+  const user = await getUserByTestUserRef(ref)
   const account = await AccountsRepository().findByUserId(user.id)
   if (account instanceof Error) throw account
   return account
 }
 
-export const getUserIdByTestUserIndex = async (index: number) => {
-  const user = await getUserByTestUserIndex(index)
+export const getUserIdByTestUserRef = async (ref: string) => {
+  const user = await getUserByTestUserRef(ref)
   return user.id
 }
 
-export const getAccountIdByTestUserIndex = async (index: number) => {
-  const account = await getAccountByTestUserIndex(index)
+export const getAccountIdByTestUserRef = async (ref: string) => {
+  const account = await getAccountByTestUserRef(ref)
   return account.id
 }
 
-export const getDefaultWalletIdByTestUserIndex = async (index: number) => {
-  const account = await getAccountByTestUserIndex(index)
+export const getDefaultWalletIdByTestUserRef = async (ref: string) => {
+  const account = await getAccountByTestUserRef(ref)
   return account.defaultWalletId
 }
 
-export const getDefaultWalletByTestUserIndex = async (index: number) => {
-  const account = await getAccountByTestUserIndex(index)
-  const user = await WalletsRepository().findById(account.defaultWalletId)
+export const getDefaultWalletIdByRole = async (role: string) => {
+  const entry = adminUsers.find((item) => item.role === role)
+  const user = await UsersRepository().findByPhone(entry?.phone as PhoneNumber)
   if (user instanceof Error) throw user
-  return user
+  const account = await AccountsRepository().findByUserId(user.id)
+  if (account instanceof Error) throw account
+  return account.defaultWalletId
 }
 
-export const getUserRecordByTestUserIndex = async (index: number) => {
-  const entry = yamlConfig.test_accounts[index]
+export const getUserRecordByTestUserRef = async (ref: string) => {
+  const entry = yamlConfig.test_accounts.find((item) => item.ref === ref)
   const phone = entry.phone as PhoneNumber
 
   return User.findOne({ phone }) as UserRecord
 }
 
+const adminUsers = [
```

Comment:
Should we consolidate the config for users to be either all yaml or all json instead of split across default.yml and test/helpers/user.ts

Reply:
probably. this can be done in another PR (working on the adminUsers for this PR here which is why I've refactoring this for now)

test_accounts is also used in prod. so the current default.yaml also serve as a documentation which is helpful I think until we have some documentation around this as well

Reply:
sounds good